linera-core/src/chain_worker/state.rs

```diff
@@ -339,13 +342,26 @@ where
         }
 
         let pending_blobs = &self.chain.manager.get().pending_blobs;
-        Ok(self
+        let tasks = self
             .recent_hashed_blobs
             .subtract_cached_items_from::<_, Vec<_>>(required_blob_ids, |id| id)
             .await
             .into_iter()
             .filter(|blob_id| !pending_blobs.contains_key(blob_id))
```

Comment:
In addition to the chain manager's pending blobs, should we also check the cache?

Reply:
What cache? 🤔 We are already checking the `recent_hashed_blobs` cache here

Reply:
Ah, right, sorry! :facepalm: